src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -201,153 +193,164 @@ public boolean canBeRotated() {
+
+            if (itemIn.stackSize - 1 <= 0) {
+                itemOut = itemIn.copy();
+                itemIn = null;
             } else {
-                test = false;
+                itemOut = itemIn.copy();
+                itemOut.stackSize = 1;
+                itemIn.stackSize = itemIn.stackSize - 1;
             }
+            if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
+            if (itemOut.stackSize == 0) itemOut = null;
 
+            inventory.setInventorySlotContents(0, itemIn);
+            inventory.setInventorySlotContents(1, itemOut);
 
-            if (ticksRemaining <= 0 && machineActive) {
-
-                if (test == false) {
-                    System.out.println(test);
-                    ticksRemaining = 0;
-
-                    if (worldObj.isRemote)
-                        return;
+            ticksRemaining = 200;
+            machineActive = true;
+            this.markForUpdate();
+            this.markDirty();
+        }
 
-                    ItemStack processItem = inventory.getStackInSlot(1);
-                    if (processItem == null) {
-                        return;
-                    }
+        if (inventory.getStackInSlot(1) != null && ticksRemaining <= 0) {
 
-                    List<Crushable> outputs = PulverizerRegistry.getOutputs(processItem);
+            ticksRemaining = 0;
 
-                    if (outputs.isEmpty())
-                        return;
+            if (worldObj.isRemote) {
+                return;
+            }
 
-                    for (int i = this.crushIndex; i < outputs.size(); i++) {
-                        this.crushIndex = i;
-                        Crushable crushable = outputs.get(this.crushIndex);
+            ItemStack processItem = inventory.getStackInSlot(1);
+            if (processItem == null) {
+                return;
+            }
 
-                        ItemStack outItem = crushable.output.copy();
-                        float itemChance = crushable.chance;
-                        boolean itemFortune = crushable.luckMultiplier == 1.0f;
+            List<Crushable> outputs = PulverizerRegistry.getOutputs(processItem);
 
-                        if (crushRNG == -1) crushRNG = rnd.nextFloat();
+            if (outputs.isEmpty())
+                return;
 
-                        if (itemFortune)
-                            itemChance = itemChance + fortuneMultiplier;
+            for (int i = this.crushIndex; i < outputs.size(); i++) {
+                this.crushIndex = i;
+                Crushable crushable = outputs.get(this.crushIndex);
 
-                        outItem.stackSize = (int) Math.round(Math.floor(itemChance) + crushRNG * itemChance % 1);
-                        if (outItem.stackSize == 0) outItem.stackSize = 1;
+                ItemStack outItem = crushable.output.copy();
+                float itemChance = crushable.chance;
+                boolean itemFortune = crushable.luckMultiplier == 1.0f;
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1244)